### PR TITLE
build: add a workflow to merge upgrade PRs

### PR DESCRIPTION
**Pull request by iamsobanjaved**
As we are automating simpler python upgrade PRs, we want to use tubular scripts to merge all eligible upgrade PRs automatically.

### CODE REVIEW

**Review by awais786 — APPROVED** [review]

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (nedbat) [inline .github/workflows/merge-python-requirements-upgrade-prs.yml:1]: Where possible, please use the [GitHub Action best practices](https://openedx.atlassian.net/wiki/spaces/AC/pages/3548282937/GitHub+Actions+best+practices). (and/or comment on them where they don't seem right to you...)

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (nedbat) [inline .github/workflows/merge-python-requirements-upgrade-prs.yml:31]: I made some review comment on the `get_ready_to_merge_prs.py` script: https://github.com/openedx/tubular/commit/b00077e3fe0724ee93d2d69918441d9ae4b1e3dd

Is there a reason to have that script in the tubular repo if it's used from here?

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (awais786) [inline .github/workflows/merge-python-requirements-upgrade-prs.yml:92]: For upgrade prs verify all required and optional checks pass.

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (iamsobanjaved) [reply .github/workflows/merge-python-requirements-upgrade-prs.yml]: We were using a few other scripts from that repo, so just hosted ours there too, otherwise was no specific reason.

**Comment by iamsobanjaved** [comment]
Sample check runs can be seen here: https://github.com/openedx/.github/actions/runs/3967072478

**Comment by awais786** [comment]
@nedbat please review

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (nedbat) [inline .github/workflows/merge-python-requirements-upgrade-prs.yml:5]: A link to cron.guru will help people understand this notation.

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (nedbat) [inline .github/workflows/merge-python-requirements-upgrade-prs.yml:36]: I thought get_ready_to_merge_prs.py was written for this action.  Why do we need to delete the first two lines of output?  I think it would be better to keep this script in this repo, and to make it more specific to its purpose.

**Review by nedbat — APPROVED** [review]
I don't want to hold up this PR, but please keep a few comment in mind if there is further work on this.

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (iamsobanjaved) [reply .github/workflows/merge-python-requirements-upgrade-prs.yml]: Yeah, moved here. Will create a PR in tubular to remove from there.

File: .github/workflows/merge-python-requirements-upgrade-prs.yml
Comment (iamsobanjaved) [reply .github/workflows/merge-python-requirements-upgrade-prs.yml:7]: Thanks, added.

**Comment by awais786** [comment]
@kdmccormick @feanil Please review this PR. If every thing seems fine please merge it also.

**Comment by feanil** [comment]
@awais786 a couple of questions for you guys so I can better understand the change:

* What's the motivation behind having this be a separate check instead of just being part of the workflow of the Python Requirements?
* Why do our own merging logic instead of using the built in Auto-merge functionality that GitHub provides?

**Comment by iamsobanjaved** [comment]
@feanil thanks for looking into this.

> What's the motivation behind having this be a separate check instead of just being part of the workflow of the Python Requirements?

Because we want CI checks to be passing before merging the pull request, we tried to find any trigger to run an action after successful CI checks but nothing worked.

> Why do our own merging logic instead of using the built in Auto-merge functionality that GitHub provides?

We were facing issues dealing with the checks which are not in the branch protection rules but are still necessary to pass, none of the unit test shards is marked as required in credentials. All solutions were merging with just required checks as green so we opted for our own code.

**Comment by awais786** [comment]
@feanil any update on this ?

**Pull request merged**
